### PR TITLE
resolve: make resolved exit cleanly

### PR DESCRIPTION
**Pull request by yuwata**
split-out of #30049.

### CODE REVIEW

**Comment by bluca** [comment]
Is this new @mrc0mmand ?

```
06:38:27 FAIL: test_address_static (__main__.NetworkdNetworkTests)
06:38:27 ----------------------------------------------------------------------
06:38:27 Traceback (most recent call last):
06:38:27   File "./test/test-network/systemd-networkd-tests.py", line 2614, in test_address_static
06:38:27     self.check_nftset('ifindex', 'dummy98')
06:38:27   File "./test/test-network/systemd-networkd-tests.py", line 1055, in check_nftset
06:38:27     self.assertRegex(output, r'.*elements = { [^}]*' + contents + r'[^}]* }.*')
06:38:27 AssertionError: Regex didn't match: '.*elements = { [^}]*dummy98[^}]* }.*' not found in 'table inet sd_test {\n\tset ifindex {\n\t\ttype iface_index\n\t}\n}'
```

**Comment by bluca** [comment]
happened here as well: https://github.com/systemd/systemd/pull/30414

**Review by bluca — APPROVED** [review]

File: src/resolve/resolved-dns-scope.c
Comment (evverx) [inline src/resolve/resolved-dns-scope.c:1437]: With this patch applied resolved no longer sends goodbye packets when it's stopped. services announced via resolved get stuck in caches until they either auto-expire or are reconfirmed after failed attempts to connect to those services.

I think this PR should be reverted.

**Comment by evverx** [comment]
Honestly I'm not even sure what was "reviewed" here. Goodbye packets were mentioned explicitly in https://github.com/systemd/systemd/pull/30049#issuecomment-1845927864 and the idea was to move that patch to a separate PR to make it easier to review it. That part of resolved isn't tested at all so pressing buttons after waiting for CI doesn't work.

File: src/resolve/resolved-dns-scope.c
Comment (yuwata) [reply src/resolve/resolved-dns-scope.c:1437]: Filed as #30421. I will take a look.

Comment (evverx) [reply src/resolve/resolved-dns-scope.c:1437]: Given that it isn't the first time PRs have gotten merged with no meaningful review I think it would make sense to enforce some sort of policy preventing those YOLO reviews followed by those YOLO merges from happening. One option would be to use CODE-OWNERS. I have no idea why PRs affecting networkd and resolved are "reviewed" with basically no knowledge of what goodbye packets or router advertisements are.

Comment (yuwata) [reply src/resolve/resolved-dns-scope.c:1437]: In this case, I forget to paste your comment in another PR. Sorry.
But, at least, with this change, the state is better than previous, as this avoids to trigger assert_return(), even if it is not critical at the time.

Anyway, let's discuss at #30421 further fix or improvement.

Comment (evverx) [reply src/resolve/resolved-dns-scope.c:1437]: >  paste your comment in another PR

I don't think it would have helped because those sloppy merges happen even with full backtraces, pcaps and so on. They can't be stopped by adding more comments. The only thing that can stop them is the CI (because the usual "review" involves just pressing the button if the CI is green) so I guess before fixing that issue it would probably make sense to revamp the mDNS testsuite (and as far as I can remember there is a PR somewhere attempting to do exactly that. plus it comes with features that should make it easier to keep track of goodbye packets among other things).

> let's discuss at https://github.com/systemd/systemd/issues/30421 further fix or improvement

Let's get past https://github.com/systemd/systemd/pull/30049 first. It should help to prevent sloppy merges too.

**Pull request merged**
